Formula/tailscale.rb

```diff
@@ -0,0 +1,35 @@
+class Tailscale < Formula
+  desc "Easiest, most secure way to use WireGuard and 2FA"
+  homepage "https://tailscale.com"
+  url "https://github.com/tailscale/tailscale.git",
+      tag:      "v1.16.2",
+      revision: "b56ba20549aedc0e56d09dca2552ca62421b2cbd"
+  license "BSD-3-Clause"
+
+  depends_on "go" => :build
+
+  def install
+    vars=Utils.safe_popen_read("./build_dist.sh", "shellvars")
+    ldflags = %W[
+      -s -w
+      -X tailscale.com/version.Long=#{vars.match(/VERSION_LONG=\"(.*)\"/)[1]}
+      -X tailscale.com/version.Short=#{vars.match(/VERSION_SHORT=\"(.*)\"/)[1]}
+      -X tailscale.com/version.GitCommit=#{vars.match(/VERSION_GIT_HASH=\"(.*)\"/)[1]}
+    ].join(" ")
+    system "go", "build", *std_go_args(ldflags: ldflags), "tailscale.com/cmd/tailscale"
+    system "go", "build", *std_go_args(ldflags: ldflags), "-o", bin/"tailscaled", "tailscale.com/cmd/tailscaled"
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/tailscale version")
+    pid = fork do
+      system bin/"tailscaled", "-tun=userspace-networking", "-socket=#{testpath}/tailscaled.socket"
+    end
+
+    sleep 2
+    assert_match "Logged out.", shell_output("#{bin}/tailscale --socket=#{testpath}/tailscaled.socket status", 1)
+  ensure
+    Process.kill("TERM", pid)
```

Comment:
I think this shouldn't be necessary any more and `brew test` should clean it up automatically. If this doesn't happen: let me know.

Reply:
@carlocab mentioned it doesn't always do this

Reply:
We had some CI runs that got locked up a while back from things like this. I think @Bo98 might've fixed it, but I'm also not sure.

Reply:
This should be prevented by https://github.com/Homebrew/brew/pull/11309.

Reply:
@MikeMcQuaid if we're not supposed to use this method anymore, is that something we can audit for?

Reply:
This seems to deadlock during `test` after doing a `fork` until the GitHub timeout kills it: https://github.com/Homebrew/homebrew-core/runs/4131278487?check_suite_focus=true#step:8:51

Reply:
Maybe try `exec` rather than `system` in `fork`?

Reply:
Isn't that what it's doing?

https://github.com/Homebrew/homebrew-core/blob/ccc808b48cadb8f53553f07d0728b26825a776f1/Formula/heartbeat.rb#L75-L78

Reply:
@carlocab Whoops, guess it doesn't work then 😭

Reply:
Oh, or it does?

@SMillerDev yeh, we should add an audit for this in `test do` I think.